ui/src/components/GraphCard.vue

```diff
@@ -134,50 +44,47 @@ export default {
                         }]
                     }
                 }
-            });
-        }
-        else if (this.isRemote) {
-               for (let i = 0; i < this.isRemote.length; i++) {
-                label.push(this.isRemote[i].key);
-                value.push(this.isRemote[i].doc_count);
-                backgroundColor.push(this.getRandomColor());
-            }
-
-            const ctx = document.getElementById('remote-chart');
+        });
 
-            new Chart(ctx, {
-                type: 'horizontalBar',
-                data: {
-                    labels: label,
-                    datasets: [{
-                        data: value,
-                        backgroundColor: backgroundColor
-                    }]
-                },
-                options: {
-                    legend: {
-                        display: false
-                    },
-                    scales: {
-                        xAxes: [{
-                            ticks: {
-                                beginAtZero: true
-                            }
-                        }]
-                    }
-                }
-            });
-        }
     },
     methods: {
         // Generate Random value of color for graph bar
         getRandomColor: function()  {
-            var letters = '0123456789ABCDEF'.split('');
-            var color = '#';
-            for (var i = 0; i < 6; i++ ) {
-                color += letters[Math.floor(Math.random() * 16)];
+            let colorList = [];
+            this.graphProp.map(() => {
+                var letters = '0123456789ABCDEF'.split('');
+                var color = '#';
+                for (var i = 0; i < 6; i++ ) {
+                    color += letters[Math.floor(Math.random() * 16)];
+                }
+                colorList.push(color);
+            })
+            
+            return colorList;
+        },
+
+        // Get Keys for graph x-axis label
+        getKeys: function() {
+            return this.graphProp.map(item => item.key);
+        },
+
+        // Get Values for graph y-axis label
+        getValues: function() {
+            return this.graphProp.map(item => item.doc_count);
+        },
+        
+        // Translate Title based on Props
+        translateTitle: function() {
+            if (this.title === 'is_remote') {
+                return this.headerText = 'Remote';
+            } else if (this.title === 'location') {
+                return this.headerText = 'Location';
+            } else if (this.title === 'employment_type') {
+                return this.headerText = 'Employment Type'
+            } else {
+                this.headerText = 'Skills';
             }
-            return color;
+            return this.headerText;
```

Comment:
I don't think `this.headerText` is being used. You can directly return the headerText.
```suggestion
            const headerTexts = {
                'is_remote': 'Remote',
                'location': 'Location',
                'employment_type': 'Employment Type',
                'skills': 'Skills',
            };
            return headerTexts[this.title];
```